src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -191,13 +191,11 @@ def battery_pool(
         battery_ids: abc.Set[int] | None = None,
         name: str | None = None,
         priority: int = -sys.maxsize - 1,
-    ) -> BatteryPoolWrapper:
-        """Return the corresponding BatteryPool instance for the given ids.
+    ) -> BatteryPool:
+        """Return a new BatteryPool instance for the given ids.
 
-        If a BatteryPool instance for the given ids doesn't exist, a new one is created
-        and returned.
-
-        The BatteryPool is wrapped in a new `BatteryPoolWrapper` instance each time.
+        If a BatteryPoolReferenceStore instance for the given battery ids doesn't exist,
```

Comment:
back-tick-quotes around `BatteryPoolReferenceStore` would be nice.

Reply:
Done in: https://github.com/frequenz-floss/frequenz-sdk-python/pull/692/commits/8d6edef480e87c12a2523f9ae009fd2bd7f0b5eb